libraries/libloader/src/main/java/org/pentaho/reporting/libraries/resourceloader/loader/URLResourceLoader.java

```diff
@@ -70,13 +70,22 @@ public ResourceKey createKey( final Object value, final Map factoryKeys ) throws
       return new ResourceKey( SCHEMA_NAME, value, factoryKeys );
     }
     if ( value instanceof String ) {
-      final String valueString = (String) value;
+      String valueString = (String) value;
+      final String encodedValue;
+      // query parameters might contain ' ' (space) that must be escaped
+      // other 'special' characters shouldn't need escaping, as they are
+      // relaxedQueryChars in Tomcat config
+      if ( valueString.contains( " " ) ) {
+        encodedValue = valueString.replace( " ", "%20" );
+      } else {
+        encodedValue = valueString;
+      }
       // the 'file' schema is defined to have double-slashes, but the JDK
       // ignores RFC 1738 in that case. So we have to check for these beasts
       // directly ..
-      if ( valueString.indexOf( "://" ) >= 0 || valueString.startsWith( "file:/" ) ) {
+      if ( encodedValue.indexOf( "://" ) >= 0 || encodedValue.startsWith( "file:/" ) ) {
```

Comment:
Can you ensure that when we are accessing to a file:// is spaces supported?

Reply:
Yes, file:// supports the usage of spaces